client/rpc_sender.go

```diff
@@ -77,33 +79,11 @@ func newSender(server string, context *base.Context, retryOpts retry.Options) (*
 // command ID to avoid reporting failure when in fact the command may have gone
 // through and been executed successfully. We retry here to eventually get
 // through with the same client command ID and be given the cached response.
-func (s *Sender) Send(_ context.Context, call proto.Call) {
-	var err error
-	for r := retry.Start(s.retryOpts); r.Next(); {
-		select {
-		case <-s.client.Healthy():
-		default:
-			log.Warningf("client %s is unhealthy; retrying", s.client)
-			continue
-		}
+func (s *rpcSender) Send(_ context.Context, call proto.Call) {
+	// Wait for it.
+	<-s.client.Healthy()
```

Comment:
Had to remove this because the assumption that all RPCs are retryable made it impossible to test certain failures. I'll put it back and remove the offending tests if this gets @mberhault's approval.